### PR TITLE
Issue #62 Faction

### PR DESCRIPTION
**Pull request by Loomie**
Affiliation is now based on the faction.

### CODE REVIEW

**Comment by Loomie** [comment]
I created a simple model that can be used in entities. Currently there is an attribute that holds a faction. That is already used in ElementEntity, CommunicationEntity and FOBEntity.

File: model/faction/src/main/kotlin/de/gleex/pltcmd/model/faction/Faction.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package de.gleex.pltcmd.model.faction
+
+/**
+ * One party in a military conflict.
+ */
+data class Faction(val name: String)
+
+val UNIDENTIFIED = Faction("unidentified")
```
Comment (Baret) [inline model/faction/src/main/kotlin/de/gleex/pltcmd/model/faction/Faction.kt:8]: As we have already talked about it: This faction is not necessary. Factions and their relations to each other are facts from the beginning of the game. There is a fixed number of factions when a game starts and each faction needs to have a relation (allied, neutral, hostile) to every other faction. Btw. **this should be verified on game start**, or can we build a `Faction` constructor that implicitly checks that? Factions would need to be singletons then.

However, when the faction of an element is "unidentified", that is the _knowledge_ of a single entity. Another entity might well be aware which faction the element belongs to. So what you tried to achieve here with `UNIDENTIFIED` comes with #104 and for simplicity's sake we should simply leave it out for now. Else we would need to remove it later on which might cause confusion or hacky code.

**Review by Baret — COMMENTED** [review]
It does not quite hit the desciption of #62 but I like the simplicity. It contains the core model for what a faction is and we can build up on this. This means we can forther down the road use it for gameplay features like the high command, friend/foe detection and so on.

But I'd like to have it without the "unidentified" faction as it feels like a hack we are going to solve with future tickets. And maybe it would be good to validate the faction relations on game start. Because what happens if you try to identify another element, you see it belongs to "Faction A" but you can not look up the relation of your own faction to that one? Shoot? Ignore them? ;)

**Comment by Loomie** [comment]
> And maybe it would be good to validate the faction relations on game start

That is for the user of the faction that creates multiple instances. I don't see how the model can do this as it does not about _all_ factions.

> or can we build a `Faction` constructor that implicitly checks that?

I don't want to mix FactionRelations into Faction - no dependency cycles! Also I think we can't do such a hard check as one Faction after the other is created and not all at once.

**Pull request merged**